sw/device/lib/base/mmio.h

```diff
@@ -288,6 +288,30 @@ inline void mmio_region_nonatomic_set_bit32(mmio_region_t base,
   mmio_region_nonatomic_set_mask32(base, offset, 0x1, bit_index);
 }
 
+/**
+ * Copies a block of memory from MMIO to main memory while ensuring that MMIO
+ * accesses are word-aligned.
```

Comment:
Nit: Maybe note that this does a read-modify-write of the boundary words?

I think a future change could even add a parameter like "copy_mode" that switches between doing byte and half word access for the boundary, and the current implementation, instead.

Reply:
Good point, done. This comment actually applies to `mmio_region_memcpy_to_mmio32` not `mmio_region_memcpy_from_mmio32`. I didn't make any changes for `mmio_region_memcpy_from_mmio32` because ignoring parts of the boundary words should not be unexpected, right?

Regarding future extensions: Agreed, but it is probably better to follow the `*_mmio{8,16,32}()`  naming convention for the public API. The actual implementation depends on what the changes will look like I guess.